### ISSUE TITLE
Feature: Downmix Stereo-Input to Mono

### ISSUE THREAD

**Issue by nathanlesage** [wulkano/kap#454]
**Platform:** macOS 10.13.4
**Kap Version:** 2.1.2.777

### Steps to reproduce

Use a two-channel audio interface and plug in a mono microphone into it. Leave the second input unused. Then record the screen using Kap and speak some instructions/explanations into your microphone.

### Current behaviour

Kap will record your audio using both Input 1 and 2, which results in only the left channel being audible, because in the right channel there has not been any input during recording; the input was unused.

### Expected behaviour

Add an option to pan both channels of a stereo-interface to center, overlapping them. This way, it doesn't matter whether you plug in your microphone into channel 1 or 2, because one of them will be silent, but won't be noticable because the other channel is also panned to center and will fill out both channels.

### Workaround

Using such a setup it's only possible by loading the rendered video into a DAW such as ProTools or CuBase and downmix the stereo-mix to a mono-mix, and then re-export it.

### My setup

My setup currently consists of the Behringer UMC 202 HD USB audio interface, recording is done via a mono condenser microphone using channel 1 of the interface. The selection of the audio interface is done using the Kap internal selector.

To get an idea of how strange this sounds, have a look at this video: https://www.youtube.com/watch?v=vStbtF4_grE

**Comment by doot0** [comment]
The audio device handling for Kap is the responsibility of the aperture library, so you should consider moving your issue to https://github.com/wulkano/aperture.

Regardless of this, I would posit that downmixing audio is out of scope for Kap.

Something that could help instead is would be adding the ability to export the audio tracks from the recording separately so you don't have to load a video file into a DAW.

**Comment by skllcrn** [comment]
Thanks @doot0. I guess you could also use something like SoundFlower or Loopback as a workaround @nathanlesage. I guess we could detect if the audio is mono before converting and use that in both channels to create stereo sound with `ffmpeg`. Thoughts @albinekb? All that being said, I agree with @doot0, I'm not sure it's within the scope of Kap. I'd argue that exporting the audio tracks separately is also is also out of scope, as that's something you can do in other programs geared toward editing. I'm all ears and would appreciate more input and example uses!

**Comment by albinekb** [comment]
I think we could detect mono audio with `ffprobe` or even before we start recording inside Aperture (the swift stuff might be able to, cc @sindresorhus) and then downmix the one channel to stereo when converting with `ffmpeg` 🤔 👍

**Comment by nathanlesage** [comment]
Yes, even a simple switcher would be great to simply say "I only want audio from Channel 1", because it would make things simpler!

I guess a mixture of GUI-elements (select the device & the number of channels) and changes to the Swift code (where only the first interface is selected and used) would be necessary.

**Comment by Mortiferr** [comment]
+1 for summing to mono. I can see the argument for it being out of scope for Kap, however. But this would be useful as someone else who uses an interface and mic which only records in one input. Yes, you _could_ export to Logic, ProTools, etc., it would be awfully convenient to be able to directly export and upload wherever if desired.

**Comment by albinekb** [comment]
@skllcrn what about adding an option in the settings under the mix to "mix mono to stereo"?
